metricflow/test/fixtures/model_fixtures.py

```diff
@@ -249,3 +249,15 @@ def extended_date_semantic_model(mf_test_session_state: MetricFlowTestSessionSta
     )
     assert model_build_result.model
     return SemanticModel(model_build_result.model)
+
+
+@pytest.fixture(scope="session")
+def self_referential_model(template_mapping: Dict[str, str]) -> UserConfiguredModel:
```

Comment:
Curious what you mean by self-referential?

Reply:
Paul and I came up with a name offline. The most applicable guidance is to name the model in regards to the tests it is relevant to. So the name will become `data_warehouse_validation_test_model`